File: src/3349_AdjacentIncreasingSubarraysDetectionI/3349.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Boolean}
def has_increasing_subarrays(nums, k)
  (0...k).map{|n| reducer( nums[n..].each_slice(k).filter{|a| a.size == k}.map{|a| strict_sort(a, k) }, false) }.any?
end

def strict_sort(nums, k) 
  return nums == nums.sort && nums.size == nums.uniq.size 
end

def reducer(nums, prev)
  return false if nums == []
  return true if nums[0] && prev 
  return reducer(nums[1..], nums[0])
end

File: src/3349_AdjacentIncreasingSubarraysDetectionI/3349.go
func hasIncreasingSubarrays(nums []int, k int) bool {
    if k == 1 { return true }

    prev := false 
    for i:=0; i < k; i++ {
        curr := true 
        for j:=i; j+k <= len(nums); j+=k {
            curr = isSortedStrict(nums, j, j+k-1)
            if prev && curr { return true }
            prev = curr
        }

        prev = false
    }

    return false
}

func isSortedStrict(nums []int, a int, b int) bool {
    for i := a ; i < b; i++ {
        if nums[i] >= nums[i+1] { return false }
    } 
    return true 
}